controllers/datafoundation.go

```diff
@@ -271,12 +273,21 @@ func (r *dataFoundationReconciler) reconcileStorageCluster() error {
 			return err
 		}
 
-		// Convert the desired size to the device set count based on the underlaying OSD size
-		desiredDeviceSetCount := int(math.Ceil(float64(r.spec.UsableCapacityInTiB) / templates.OSDSizeInTiB))
+		// get the desired capacity in TiB planned to be consumed by user in range 1-4TiB
+		desiredOSDSizeInTiB := math.Min(math.Ceil(float64(r.spec.UsableCapacityInTiB)), templates.VerticalScalerUpperBoundInTiB)
 
-		// Get the storage device set count of the current storage cluster
+		// Convert the desired size to the device set count based on the underlying OSD size
+		desiredDeviceSetCount := int(math.Ceil(float64(r.spec.UsableCapacityInTiB)) / desiredOSDSizeInTiB)
```

Comment:
Very similar questions here.
Why are we basing desired state on actual state (r.spec.*)?

Also, I believe there is a bug here. Converting to `int` will lose the friction not round it up. The actual statement should be
```suggestion
		desiredDeviceSetCount := math.Ceil(float64(r.spec.UsableCapacityInTiB) / desiredOSDSizeInTiB)
```

Reply:
I guess you meant the casting should wrap the entire expression (it is after all an integer):
```Go
desiredDeviceSetCount := int(math.Ceil(float64(r.spec.UsableCapacityInTiB) / desiredOSDSizeInTiB))
```

Reply:
Yep, that makes sense